site/app/templates/calendar/Calendar.twig

```diff
@@ -78,11 +78,38 @@
         <div class="cal-legend">
             {% if show_all_cookie == 1 %}
                 {% for course in course_names %}
-                    <div name="{{course.course_name|replace({' ':''})}}" class="legend-color"></div>
-                    <p for="{{course.course_name|replace({' ':''})}}-color" class="legend-label">{{course.display_name}}</p>
+                <div 
+                    name="{{course.course_name|replace({' ':''})}}" 
+                    class="legend-color" 
+                    onclick="filter_course('{{course.course_name}},{{course.display_name}}')" 
+                    style="cursor: pointer;" 
+                    title="Click to open Calendar options form"
+                ></div>
+                <p 
+                    for="{{course.course_name|replace({' ':''})}}-color" 
+                    style="cursor: pointer;" 
+                    onclick="filter_course('{{course.course_name}},{{course.display_name}}')" 
+                    title="Click to filter course, click again to show all courses" 
+                    class="legend-label"
+                    >{{course.display_name}}
+                </p>
                 {% endfor %}
             {% else %}
-                <div name="{{calendar_course_cookie|replace({' ':''})}}" class="legend-color"></div><p for="{{calendar_course_cookie|replace({' ':''})}}-color" class="legend-label">{{calendar_course_cookie}}</p>
+                <div 
+                    name="{{calendar_course_cookie|replace({' ':''})}}" 
+                    class="legend-color" 
+                    onclick="filter_course('{{"show all"}},{{""}}')"
+                    style="cursor: pointer;" 
+                    title="Click to open Calendar options form"
+                ></div>
+                <p 
+                    for="{{calendar_course_cookie|replace({' ':''})}}-color" 
+                    style="cursor: pointer;" 
+                    onclick="filter_course('{{"show all"}},{{""}}')" 
```

Comment:
```suggestion
                    onclick="filter_course('show all', '')" 
```
Again here. I'm passing the empty string through Javascript directly rather than through Twig first.